test/Microsoft.IdentityModel.Tokens.Tests/Validation/SigningKeyValidationResultTests.cs

```diff
@@ -50,11 +61,10 @@ public static TheoryData<SigningKeyValidationTheoryData> SigningKeyValidationTes
                     new SigningKeyValidationTheoryData
                     {
                         TestId = "Valid_SecurityTokenIsPresent",
-                        ExpectedException = ExpectedException.NoExceptionExpected,
                         SecurityKey = KeyingMaterial.SymmetricSecurityKey2_256,
                         SecurityToken = new JwtSecurityToken(),
                         ValidationParameters = new ValidationParameters(),
-                        SigningKeyValidationResult = new SigningKeyValidationResult(KeyingMaterial.SymmetricSecurityKey2_256)
+                        Result = new ValidatedSigningKeyLifetime(null, null, utcNow)
```

Comment:
This test is a bit flaky. Could we do something like this instead?

        private readonly DateTime utcNow;
        
        public SigningKeyValidationResultTests()
        {
            utcNow = DateTime.UtcNow;
        }

Reply:
Here's the failed run log.
)   Microsoft.IdentityModel.Tokens.Validation.Tests.SigningKeyValidationResultTests.SecurityKey(theoryData: Valid_SecurityTokenIsPresent, NoExceptionExpected) 
      Duration: 12.2 sec

      Message: 
Microsoft.IdentityModel.TestUtils.TestException : Microsoft.IdentityModel.Tokens.Validation.Tests.SigningKeyValidationResultTests.SigningKeyValidationResultTests : Valid_SecurityTokenIsPresent


dateTime1 != dateTime2. '8/27/2024 10:18:42 PM' != '8/27/2024 10:18:44 PM'.


      Stack Trace: 
TestUtilities.AssertFailIfErrors(String testId, List`1 errors) line 371
TestUtilities.AssertFailIfErrors(CompareContext context) line 352
SigningKeyValidationResultTests.SecurityKey(SigningKeyValidationTheoryData theoryData) line 48

Reply:
We are creating the `utcNow` variable when the test starts. It is no different than creating it in the class as you suggest.
This is the same way we do in our lifetime validation tests, but because the signature validation is a way more expensive operation, the tests are running for some seconds and causing the flakiness.
What would solve this problem for good is to implement the DateTimeProvider that we discussed on Monday since we could ensure that the test receives always the same `utcNow` provided by our mock.

I'll increase our epsilon for now, until we address the root cause.